superset-frontend/src/cccs-viz/plugins/plugin-chart-application-links/src/plugin/controlPanel.ts

```diff
@@ -0,0 +1,147 @@
+  * will automatically append the `groupby` field when the query is generated.
+  *
+  * It is also possible to define custom controls by importing the
+  * necessary dependencies and overriding the default parameters, which
+  * can then be placed in the `controlSetRows` section
+  * of the `Query` section instead of a predefined control.
+  *
+  * import { validateNonEmpty } from '@superset-ui/core';
+  * import {
+  *   sharedControls,
+  *   ControlConfig,
+  *   ControlPanelConfig,
+  * } from '@superset-ui/chart-controls';
+  *
+  * const myControl: ControlConfig<'SelectControl'> = {
+  *   name: 'secondary_entity',
+  *   config: {
+  *     ...sharedControls.entity,
+  *     type: 'SelectControl',
+  *     label: t('Secondary Entity'),
+  *     mapStateToProps: state => ({
+  *       sharedControls.columnChoices(state.datasource)
+  *       .columns.filter(c => c.groupby)
+  *     })
+  *     validators: [validateNonEmpty],
+  *   },
+  * }
+  *
+  * In addition to the basic drop down control, there are several predefined
+  * control types (can be set via the `type` property) that can be used. Some
+  * commonly used examples:
+  * - SelectControl: Dropdown to select single or multiple values,
+      usually columns
+  * - MetricsControl: Dropdown to select metrics, triggering a modal
+      to define Metric details
+  * - AdhocFilterControl: Control to choose filters
+  * - CheckboxControl: A checkbox for choosing true/false values
+  * - SliderControl: A slider with min/max values
+  * - TextControl: Control for text data
+  *
+  * For more control input types, check out the `incubator-superset` repo
+  * and open this file: superset-frontend/src/explore/components/controls/index.js
+  *
+  * To ensure all controls have been filled out correctly, the following
+  * validators are provided
+  * by the `@superset-ui/core/lib/validator`:
+  * - validateNonEmpty: must have at least one value
+  * - validateInteger: must be an integer value
+  * - validateNumber: must be an intger or decimal value
+  */
+
+  // For control input types, see: superset-frontend/src/explore/components/controls/index.js
+  controlPanelSections: [
+    {
+      label: t('Query'),
+      expanded: true,
+      controlSetRows: [
+        ['adhoc_filters'],
+        [
+          {
+            name: 'columns',
+            override: {
+              mapStateToProps: (
+                state: ControlPanelState,
+                controlState: ControlState,
+              ) => {
+                const originalMapStateToProps =
+                  sharedControls?.columns?.mapStateToProps;
+                const newState =
+                  originalMapStateToProps?.(state, controlState) ?? {};
+                // @ts-ignore
+                newState.externalValidationErrors =
+                  // isRawMode({ controls }) &&
+                  ensureIsArray(controlState.value).length === 0
+                    ? [t('must have a value')]
```

Comment:
Another one of those copy and paste from an example component things.  It doesn't show up in the UI unless there is nothing chosen in the list of columns.  There isn't a lot of room for the message so it will likely wrap if the wording is too long.  I will change it to "This control must have a value." and hopefully that doesn't wrap too much / look ugly :)